### PR TITLE
Add Imagenette to TensorFlow datasets

### PR DESCRIPTION
**Pull request by cking100**
This PR adds the [imagenette](https://github.com/fastai/imagenette) dataset as part of the tensorflow datasets.

### CODE REVIEW

File: tensorflow_datasets/image/imagenette.py [hunk 1]
```diff
@@ -0,0 +1,139 @@
...
+import itertools
+import os
+
+import numpy as np
+import tensorflow as tf
+
+import tensorflow_datasets.public_api as tfds
+
+# TODO(imagenette): BibTeX citation
+_CITATION = """
```
Comment (cking100) [inline tensorflow_datasets/image/imagenette.py]: I still have to add citation as I couldn't find one on the website.

Comment (cyfra) [reply tensorflow_datasets/image/imagenette.py]: You can contact the author of the dataset and ask if they have one.

File: tensorflow_datasets/image/imagenette.py [hunk 2]
```diff
@@ -0,0 +1,139 @@
...
+    configs.append(
+        ImagenetteConfig(
+            name='%s' % (size),
+            size=size,
+            description=_DESCRIPTION.format(size=size)))
+  return configs
+
+
+class Imagenette(tfds.core.GeneratorBasedBuilder):
+  """Imagenette:A smaller subset of 10 easily classified classes from
```
Comment (cyfra) [inline tensorflow_datasets/image/imagenette.py]: please run the lint on your code:

https://github.com/tensorflow/datasets/blob/38189d93462b6e4f8fab820d416d8959c11d85e4/docs/add_dataset.md#5-check-your-code-style

File: tensorflow_datasets/image/imagenette.py [hunk 3]
```diff
@@ -0,0 +1,139 @@
...
+        supervised_keys=('image', 'label'),
+        homepage='https://github.com/fastai/imagenette',
+        citation=_CITATION,
+    )
+
+  def _split_generators(self, dl_manager):
+    """Returns SplitGenerators."""
+    size = self.builder_config.size
+
+    if size in ["320 px","160 px"]:
```
Comment (cyfra) [inline tensorflow_datasets/image/imagenette.py]: rather than repeating these lines 3 times, you could compute the "file-prefix" instead.

(file_prefix == "imagenette-160" or "imagenette-320" or "imagenette")

and then use it in the lines below.

File: tensorflow_datasets/image/imagenette.py [hunk 4]
```diff
@@ -0,0 +1,139 @@
...
+      path = dl_manager.download_and_extract('%s/imagenette.tgz' % (_URL_PREFIX))
+      train_path = os.path.join(path, "imagenette", "train")
+      val_path = os.path.join(path, "imagenette", "val")
+    else:
+      raise ValueError('Size not implemented!')
+
+    return [
+        tfds.core.SplitGenerator(
+            name=tfds.Split.TRAIN,
+            num_shards=10,  # Ignored when using a version with S3 experiment.
```
Comment (cyfra) [inline tensorflow_datasets/image/imagenette.py]: you should be able to remove this line (as S3 should be enabled).

File: tensorflow_datasets/image/imagenette.py [hunk 5]
```diff
@@ -0,0 +1,139 @@
...
+            gen_kwargs={
+                "datapath": train_path,
+            },
+        ),
+        tfds.core.SplitGenerator(
+            name=tfds.Split.VALIDATION,
+            num_shards=1,  # Ignored when using a version with S3 experiment.
```
Comment (cyfra) [inline tensorflow_datasets/image/imagenette.py]: same here.

File: tensorflow_datasets/image/imagenette_test.py [hunk 6]
```diff
@@ -0,0 +1,37 @@
...
+from __future__ import print_function
+
+from tensorflow_datasets import testing
+from tensorflow_datasets.image import imagenette
+
+
+class ImagenetteTest(testing.DatasetBuilderTestCase):
+  DATASET_CLASS = imagenette.Imagenette
+  SPLITS = {
+      "train": 4,  # Number of fake train example
```
Comment (cyfra) [inline tensorflow_datasets/image/imagenette_test.py]: your test images have fake random data (good) - but this also means that they don't really compress well (and some take over 3MB).

You can use function from testing/fake_data_utils.py 
(I see that png one is doing the "smart" thing, but the jpeg one is still generating large random images - we should fix that).

Comment (cking100) [reply tensorflow_datasets/image/imagenette_test.py]: I have used the function from testing/fake_data_utils.py to generate fake random data. The images are large cuz the size I choose was large something like 2304 x 1728. The only reason for choosing a large size because I have seen similar size images in the original dataset.

File: tensorflow_datasets/image/imagenette.py (hunk 2 above)
Comment (cking100) [reply tensorflow_datasets/image/imagenette.py]: Okay, I will do it.

File: tensorflow_datasets/image/imagenette.py (hunk 3 above)
Comment (cking100) [reply tensorflow_datasets/image/imagenette.py]: Good idea. Will do it.

**Comment by cking100** [comment]
@cyfra I have made the requested changes. Please have a look. I have also contacted the author for the citation. Now, I'm just left with https://github.com/tensorflow/datasets/pull/1261#discussion_r354002048. How should I fix it any suggestions?

**Comment by cking100** [comment]
I have also removed the big jpeg files from fake data.

File: tensorflow_datasets/image/imagenette.py (hunk 2 above)
Comment (cyfra) [reply tensorflow_datasets/image/imagenette.py]: Did it run ? (as normally it complains that the comment should fit in a single line (with a "dot" at the end).

File: tensorflow_datasets/image/imagenette.py [hunk 7]
```diff
@@ -0,0 +1,131 @@
...
+      path = dl_manager.download_and_extract(url)
+      train_path = os.path.join(path, "imagenette", "train")
+      val_path = os.path.join(path, "imagenette", "val")
+    else:
+      raise ValueError('Size not implemented!')
+
+    return [
+        tfds.core.SplitGenerator(
+            name=tfds.Split.TRAIN,
+            num_shards=10,
```
Comment (cyfra) [inline tensorflow_datasets/image/imagenette.py]: please remove num_shards

File: tensorflow_datasets/image/imagenette.py [hunk 8]
```diff
@@ -0,0 +1,131 @@
...
+  """BuilderConfig for Imagenette."""
+
+  def __init__(self, **kwargs):
+    super(ImagenetteConfig, self).__init__(
+        version=tfds.core.Version('0.1.0'), **kwargs)
+
+
+def _make_builder_configs():
+  configs = []
+  for variant in _VARIANTS:
```
Comment (cyfra) [inline tensorflow_datasets/image/imagenette.py]: BTW -  having the config name "imagenette-160.tgz" doesn't look nice 
(as people will have to say:

tfds.load("imagenette/imagenette-160.tar")

Could you have the config named 160px instead ?

Comment (cking100) [reply tensorflow_datasets/image/imagenette.py]: So use 160px instead of imagenette-160.tgz as the `name` parameter in `ImagenetteConfig`?

File: tensorflow_datasets/image/imagenette.py (hunk 7 above)
Comment (cking100) [reply tensorflow_datasets/image/imagenette.py]: Sure.

File: tensorflow_datasets/image/imagenette.py (hunk 2 above)
Comment (cking100) [reply tensorflow_datasets/image/imagenette.py]: I haven't run it on the whole tensorflow_dataset/image directory but I have checked it with the files that I have added and it is given 10/10 on both the files that I have added.

Comment (cking100) [reply tensorflow_datasets/image/imagenette.py]: image

**Comment by cking100** [comment]
I have made the changes. Please have a look. Thanks

File: tensorflow_datasets/image/imagenette.py (hunk 1 above)
Comment (rickwierenga) [reply tensorflow_datasets/image/imagenette.py]: https://github.com/fastai/imagenette/issues/16

**Comment by guillaumelorre28** [comment]
Hello everyone,
the Imagenette dataset have changed:
"The Imagenette and Imagewoof datasets have recently (Dec 6 2019) changed. They now have a 70/30 train/valid split." (from https://github.com/fastai/imagenette)
The imagenette tensorflow dataset has still the small validation set.
It is impossible to compare to the fastai leaderboard using this dataset.

Would it be possible to update it using the new version?

**Comment by Conchylicultor** [comment]
@guillaumelorre28 created #1679 to track. Please don't hesitate to send a PR with the new version. Do you think we should keep the two versions or only the new one ?
We could keep the two versions by using `BuilderConfig`.
